test/commands/scanner/rule/add.test.ts

```diff
@@ -9,43 +9,147 @@ import path = require('path');
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/sfdx-scanner', 'add');
 
+const CATALOG_OVERRIDE = 'AddTestPmdCatalog.json';
+const CUSTOM_PATH_OVERRIDE = 'AddTestCustomPaths.json';
+
+// Delete any existing JSONs associated with the tests so they run fresh each time.
+if (fs.existsSync(path.join(SFDX_SCANNER_PATH, CATALOG_OVERRIDE))) {
+  fs.unlinkSync(path.join(SFDX_SCANNER_PATH, CATALOG_OVERRIDE));
+}
+if (fs.existsSync(path.join(SFDX_SCANNER_PATH, CUSTOM_PATH_OVERRIDE))) {
+  fs.unlinkSync(path.join(SFDX_SCANNER_PATH, CUSTOM_PATH_OVERRIDE));
+}
+
+let addTest = test.env({PMD_CATALOG_NAME: CATALOG_OVERRIDE, CUSTOM_PATH_FILE: CUSTOM_PATH_OVERRIDE});
+
 
 describe('scanner:rule:add', () => {
   describe('E2E', () => {
-    describe('Happy Path', () => {
-      const CATALOG_OVERRIDE = 'AddTestPmdCatalog.json';
-      const CUSTOM_PATH_OVERRIDE = 'AddTestCustomPaths.json';
+    const myLanguage = 'apex';
+    describe('Test Case: Adding individual JARs', () => {
+      // Create four temporary directories, each having a single JAR.
+      const tmpDir1 = fs.mkdtempSync(path.join(os.tmpdir(), 'foo-'));
+      const tmpJar1 = path.join(tmpDir1, 'bar1.jar');
+      fs.writeFileSync(tmpJar1, 'In the Age of Ancients, the world was unformed and shrouded by fog.');
```

Comment:
Your text is too interesting and I _had_ to find the source!
https://en.wikipedia.org/wiki/Dark_Souls